docs/content/dagster-cloud/insights/integrating-snowflake.mdx

```diff
@@ -0,0 +1,98 @@
+---
+title: "Integrating Snowflake with Dagster Cloud Insights | Dagster Docs"
+description: "Integrating external metrics with Dagster Insights."
+
+platform_type: "cloud"
+---
+
+# Integrating Snowflake with Dagster Cloud Insights
+
+External metrics, such as Snowflake credits, can be integrated into the Dagster Insights UI. The [`dagster-cloud`](https://pypi.org/project/dagster-cloud/) package contains utilities for capturing and submitting external metrics about data operations to Dagster Cloud via an API.
+
+If you use the [Snowflake Resource](/libraries/dagster-snowflake) to query Snowflake, use this guide to integrate Snowflake metrics into the Insights UI. For instructions on integrating usage of dbt models which run in Snowflake, see [Integrating Snowflake + dbt with Dagster Cloud Insights](/dagster-cloud/insights/integrating-snowflake-and-dbt).
+
+---
+
+## Prerequisites
+
+To complete the steps in this guide, you'll need:
+
+- A Dagster Cloud account on the Enterprise plan
+- Access to the [Dagster Cloud Insights feature](/dagster-cloud/insights)
+- To install the following libraries:
+
+  ```shell
+  pip install dagster dagster-cloud dagster-snowflake
+  ```
+
+  **Note**: If you already have `dagster-cloud` installed, **make sure you're using version 1.5.8 or newer**.
+
+---
+
+## Step 1: Replace your Snowflake resource(s)
+
+The first step is to replace your existing Snowflake resource(s) with the <PyObject module="dagster_cloud" object="InsightsSnowflakeResource" /> resource. This resource is a drop-in replacement for the <PyObject module="dagster_snowflake" object="SnowflakeResource" /> resource, but it also emits Snowflake usage metrics to the Dagster Cloud Insights API.
+
+```python
```

Comment:
Since the creds for this resource will have to have access to a specific table, we should say so here. I mentioned in a comment in the next section that the table access should be a prereq, so it could be something like:

"If needed, update the credentials used by the resource to a Snowflake user that has access to the `query_history` table."